app/models/manageiq/providers/openstack/inventory/collector/target_collection.rb

```diff
@@ -104,19 +105,31 @@ def orchestration_stacks
     []
   end
 
-  def get_orchestration_stack(stack_id, tenant_id)
-    tenant = memoized_get_tenant(tenant_id)
-    safe_get { @os_handle.detect_orchestration_service(tenant.try(:name)).stacks.get(stack_id) }
+  def indexed_all_orchestration_stacks
+    @indexed_all_orchestration_stacks ||= all_orchestration_stacks.index_by(&:id)
+  end
+
+  def get_orchestration_stack(stack_id, _tenant_id = nil)
+    # TODO fog needs to implement /v1/{tenant_id}/stacks/{stack_identity} call, right now the only supported call
```

Comment:
@aufi this is the TODO you should focus on. So we are able to fetch orchestration stack with :ems_ref and it searches in nested stack too. So adding `/v1/{tenant_id}/stacks/{stack_identity} ` to fog

Reply:
the workaround now is that we need to fetch all nested orchestration stacks, instead of fetching the specific ones involved in targeted refresh. So this one call is making targeted refresh much slower (in reality, it would be quicker if we have e.g. 1000 stack targets, which is logic I have implemented in Azure, doing list vs. get based on number of targets)

Reply:
Got it, thanks!